js/dropdown/shared/Dropdown.svelte

```diff
@@ -92,9 +93,19 @@
 	}
 
 	$: {
-		if (choices !== old_choices || input_text !== old_input_text) {
-			filtered_indices = handle_filter(choices, input_text);
+		if (choices !== old_choices) {
+			set_input_text();
 			old_choices = choices;
+			filtered_indices = handle_filter(choices, input_text);
+			if (!allow_custom_value && filtered_indices.length > 0) {
```

Comment:
A little bit of code duplication from below, but I think its easier to follow this way